File: lua/8-struktury.lua
#!/bin/lua

samochod={}
marka=1
rok=2

function pokaz(s)
	print("marka:", s["marka"], s[marka])
	print("rok produkcji:", s["rok"], s[rok])
end

samochod["marka"]="volvo v40"
samochod["rok"]=2010

samochod[marka]="volvo v40"
samochod[rok]=2010
	
pokaz(samochod)

File: perl/8-struktury.pl
#!/bin/perl

%volvo=( 
	'marka' => 'volvo v40', 
	'rok_produkcji' => 2010);

sub pokaz
{
	my (%h) = @_;
	print "marka: $h{'marka'}\n";
	print "rok produkcji: $h{'rok_produkcji'}\n";

}

sub hash_tab 
{
	my %h = @_;
	foreach my $key (keys %h) {
		print "key=$key value=$h{$key}\n";
	}
}


pokaz(%volvo);
hash_tab(%volvo);
